### PR TITLE
Preserve ipython magics.

### PR DESCRIPTION
**Pull request by girip11**
Addresses bug #445

Changes to be committed:
	modified:   nbqa/handle_magics.py
	modified:   setup.cfg

### CODE REVIEW

**Comment by girip11** [comment]
closes #445

**Comment by girip11** [comment]
Tested on pymc3 notebooks as well as the notebooks present in the [henryiii/python-compiled-minicourse](https://github.com/henryiii/python-compiled-minicourse).

**Comment by girip11** [comment]
This PR does not address #455. But changes made in this PR makes it easier to fix #455.

**Comment by MarcoGorelli** [comment]
> This PR does not address #455. But changes made in this PR makes it easier to fix #455.

Awesome! So this PR changes e.g.
```
flake8_version = !pip list 2>&1 | grep flake8
```
into
````
type(0xCDB9ABB2)  # flake8_ver 0xCDB9ABB2
```
, which won't throw any errors but which may still throw a false-positive (as per #455).

Looks good - do you have an idea already about how to transform the code to take care of #455?

File: tests/test_ipython_magics.py [hunk 1]
```diff
@@ -123,69 +124,116 @@ def test_ignore_cells(
...
+def _validate_magics_with_black(before: List[str], after: List[str]) -> bool:
+    """
+    Validate the state of the notebook before and after running nbqa with black.
+
+    Parameters
+    ----------
+    before
+        Notebook contents before running nbqa with autoflake
+    after
+        Notebook contents after running nbqa with autoflake
```
Comment (MarcoGorelli) [inline tests/test_ipython_magics.py:136]: I'm a bit confused here - the summary says "before and after running nbqa with black", but the param descriptions refer to autoflake

**Review by MarcoGorelli — APPROVED** [review]

File: tests/test_ipython_magics.py (hunk 1 above)
Comment (girip11) [reply tests/test_ipython_magics.py:136]: Good catch. copy pasted. Forgot to change. I will change it to black.

Comment (MarcoGorelli) [reply tests/test_ipython_magics.py:136]: cool, feel free to merge after that

**Comment by girip11** [comment]
> - do you have an idea already about how to transform the code to take care of #455?

Yeah I have a rough idea to handle this issue. I will comment on the issue about the approach that I have.

File: tests/test_ipython_magics.py (hunk 1 above)
Comment (girip11) [reply tests/test_ipython_magics.py]: Sure

**Pull request merged**